internal/host/validator.go

```diff
@@ -342,6 +343,49 @@ func (v *validator) printCompatibleWithClusterPlatform(c *validationContext, sta
 	}
 }
 
+func isDiskEncryptionRelevantForHost(c *validationContext) bool {
+	switch *c.cluster.DiskEncryption.EnableOn {
+	case models.DiskEncryptionEnableOnAll:
+		return true
+	case models.DiskEncryptionEnableOnMasters:
+		return c.host.Role == models.HostRoleMaster || c.host.Role == models.HostRoleBootstrap
+	case models.DiskEncryptionEnableOnWorkers:
+		return c.host.Role == models.HostRoleWorker
+	default: // models.DiskEncryptionEnableOnNone
+		return false
+	}
+}
+
+func (v *validator) hasMinRequirementsForDiskEncryption(c *validationContext) ValidationStatus {
+
+	if c.infraEnv != nil || reflect.DeepEqual(c.cluster.DiskEncryption, &models.DiskEncryption{}) || !isDiskEncryptionRelevantForHost(c) {
+		return ValidationSuccessSuppressOutput
+	}
+
+	if c.inventory == nil {
+		return ValidationPending
+	}
+
+	if *c.cluster.DiskEncryption.Mode != models.DiskEncryptionModeTpmv2 {
+		return ValidationSuccess
+	}
+
+	return boolValue(c.inventory.TpmVersion == models.InventoryTpmVersionNr20)
+}
+
+func (v *validator) printHasMinRequirementsForDiskEncryption(c *validationContext, status ValidationStatus) string {
+	switch status {
+	case ValidationSuccess:
+		return fmt.Sprintf("Host's installation disk will be encrypted using %s", *c.cluster.DiskEncryption.Mode)
+	case ValidationFailure:
+		return fmt.Sprintf("The host's TPM version is not supported, expected-version: %s, actual-version: %s", models.InventoryTpmVersionNr20, c.inventory.TpmVersion)
```

Comment:
1. I would write separate messages for wrong version and for "TPM not detected"
2. Are there any validations for tang?

Reply:
1. How will I defer between the 2 cases? in both cases, the validation's result is `failure`.
2. No, but the URLs validity is checked upon cluster-creation/update

Reply:
1. I see that some of the print methods in this file have "if"s for this purpose.  Not so efficient but I guess that's the framework.
2. Do we want to do some ping periodically to see that it's reachable?  (Can be in a separate task)

Reply:
2. I don't think we can, if the tang server is accessible by anyone then anyone can decrypt the disk right?! AFAIK the server is on the local machine and not exposed via public IP (maybe only in private network)

Reply:
Fixed 1.

Reply:
I meant to have the agent ping it.  It's a bigger task - I would just open a JIRA for later.

Reply:
But how will the agent deliver this "ping info" to the service?

Reply:
https://issues.redhat.com/browse/MGMT-7741